packages/features/ee/payments/api/mercadopago-webhook.ts

```diff
@@ -0,0 +1,296 @@
+import type { Prisma } from "@prisma/client";
+import type { NextApiRequest, NextApiResponse } from "next";
+import getRawBody from "raw-body";
+import * as z from "zod";
+
+import MercadoPago from "@calcom/app-store/mercadopago/lib/MercadoPago";
+import { getMercadoPagoAppKeys } from "@calcom/app-store/mercadopago/lib/getMercadoPagoAppKeys";
+import type { MercadoPagoUserCredentialSchema } from "@calcom/app-store/mercadopago/lib/mercadoPagoCredentialSchema";
+import { mercadoPagoCredentialSchema } from "@calcom/app-store/mercadopago/lib/mercadoPagoCredentialSchema";
+import EventManager from "@calcom/core/EventManager";
+import { sendScheduledEmails } from "@calcom/emails";
+import { getCalEventResponses } from "@calcom/features/bookings/lib/getCalEventResponses";
+import { handleConfirmation } from "@calcom/features/bookings/lib/handleConfirmation";
+import { isPrismaObjOrUndefined, parseRecurringEvent } from "@calcom/lib";
+import { IS_PRODUCTION } from "@calcom/lib/constants";
+import { getErrorFromUnknown } from "@calcom/lib/errors";
+import { HttpError as HttpCode } from "@calcom/lib/http-error";
+import { getTranslation } from "@calcom/lib/server/i18n";
+import prisma, { bookingMinimalSelect } from "@calcom/prisma";
+import { BookingStatus } from "@calcom/prisma/enums";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+async function getEventType(id: number) {
+  return prisma.eventType.findUnique({
+    where: {
+      id,
+    },
+    select: {
+      recurringEvent: true,
+      requiresConfirmation: true,
+    },
+  });
+}
+
+export async function handlePaymentSuccess(payload: z.infer<typeof eventSchema>) {
+  // We have MercadoPago payment id from the payload
+  const { client_id, client_secret } = await getMercadoPagoAppKeys();
```

Comment:
Unlike Stripe and Paypal webhooks, here the event itself is not going to help us retrieve the payment/booking from Cal.com database. We will need to query the full payment from MercadoPago so that we can know which Cal.com payment/booking is associated with.